Your header seems to mismatch the implementation
```quote
 Error: /home/runner/work/mixxx/mixxx/src/effects/backends/builtin/pitchshifteffect.cpp:5:23: error: function previously declared with an implicit exception specification redeclared with an explicit exception specification [-Werror,-Wimplicit-exception-spec-mismatch]
PitchShiftGroupState::~PitchShiftGroupState() noexcept {
                      ^
/home/runner/work/mixxx/mixxx/src/effects/backends/builtin/pitchshifteffect.h:29:13: note: previous declaration is here
    virtual ~PitchShiftGroupState();
```

--- a/src/effects/backends/builtin/pitchshifteffect.cpp
+++ b/src/effects/backends/builtin/pitchshifteffect.cpp
@@ -5,7 +5,7 @@
 PitchShiftGroupState::PitchShiftGroupState(
         const mixxx::EngineParameters& engineParameters)
         : EffectState(engineParameters) {
-    initializeBuffer();
+    initializeBuffer(engineParameters);
     audioParametersChanged(engineParameters);
 }
 
@@ -20,9 +20,12 @@ PitchShiftGroupState::~PitchShiftGroupState() {
     m_pRubberBand->reset();
 }
 
-void PitchShiftGroupState::initializeBuffer() {
-    m_retrieveBuffer[0] = SampleUtil::alloc(MAX_BUFFER_LEN);
-    m_retrieveBuffer[1] = SampleUtil::alloc(MAX_BUFFER_LEN);
+void PitchShiftGroupState::initializeBuffer(
+        const mixxx::EngineParameters& engineParameters) {
+    m_retrieveBuffer[0] = SampleUtil::alloc(
+            engineParameters.framesPerBuffer());
+    m_retrieveBuffer[1] = SampleUtil::alloc(
+            engineParameters.framesPerBuffer());
 }
 
 void PitchShiftGroupState::audioParametersChanged(

--- a/src/effects/backends/builtin/pitchshifteffect.cpp
+++ b/src/effects/backends/builtin/pitchshifteffect.cpp
@@ -12,12 +12,6 @@ PitchShiftGroupState::PitchShiftGroupState(
 PitchShiftGroupState::~PitchShiftGroupState() {
     SampleUtil::free(m_retrieveBuffer[0]);
     SampleUtil::free(m_retrieveBuffer[1]);
-
-    VERIFY_OR_DEBUG_ASSERT(m_pRubberBand) {
-        return;
-    }
-
-    m_pRubberBand->reset();
 }
 
 void PitchShiftGroupState::initializeBuffer(

--- a/src/effects/backends/builtin/pitchshifteffect.cpp
+++ b/src/effects/backends/builtin/pitchshifteffect.cpp
@@ -2,7 +2,14 @@
 
 #include "util/sample.h"
 
-PitchShiftGroupState::~PitchShiftGroupState() noexcept {
+PitchShiftGroupState::PitchShiftGroupState(
+        const mixxx::EngineParameters& engineParameters)
+        : EffectState(engineParameters) {
+    initializeBuffer();
+    audioParametersChanged(engineParameters);
+}
+
+PitchShiftGroupState::~PitchShiftGroupState() {
     SampleUtil::free(m_retrieveBuffer[0]);
     SampleUtil::free(m_retrieveBuffer[1]);
 
@@ -13,9 +20,25 @@ PitchShiftGroupState::~PitchShiftGroupState() noexcept {
     m_pRubberBand->reset();
 }
 
+void PitchShiftGroupState::initializeBuffer() {
+    m_retrieveBuffer[0] = SampleUtil::alloc(MAX_BUFFER_LEN);
+    m_retrieveBuffer[1] = SampleUtil::alloc(MAX_BUFFER_LEN);
+}
+
+void PitchShiftGroupState::audioParametersChanged(
+        const mixxx::EngineParameters& engineParameters) {
+    m_pRubberBand = std::make_unique<RubberBand::RubberBandStretcher>(
+            engineParameters.sampleRate(),
+            engineParameters.channelCount(),
+            RubberBand::RubberBandStretcher::OptionProcessRealTime);
+
+    m_pRubberBand->setMaxProcessSize(engineParameters.framesPerBuffer());
+    m_pRubberBand->setTimeRatio(1.0);
+};
+
 // static
 QString PitchShiftEffect::getId() {
-    return "org.mixxx.effects.pitchshift";
+    return QStringLiteral("org.mixxx.effects.pitchshift");
 }
 
 //static
@@ -35,11 +58,9 @@ EffectManifestPointer PitchShiftEffect::getManifest() {
     pitch->setName(QObject::tr("Pitch"));
     pitch->setShortName(QObject::tr("Pitch"));
     pitch->setDescription(QObject::tr(
-            "The new pitch of a sound."));
+            "The pitch shift applied to the sound."));
     pitch->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
-    pitch->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
     pitch->setDefaultLinkType(EffectManifestParameter::LinkType::Linked);
-    pitch->setDefaultLinkInversion(EffectManifestParameter::LinkInversion::Inverted);
     pitch->setNeutralPointOnScale(0.0);
     pitch->setRange(-1.0, 0.0, 1.0);
 
@@ -63,13 +84,13 @@ void PitchShiftEffect::processChannel(
 
     const double pitchParameter = m_pPitchParameter->value();
 
-    double pitch = 1.0;
-
-    if (pitchParameter < 0.0) {
-        pitch += pitchParameter / 2.0;
-    } else {
-        pitch += pitchParameter;
-    }
+    const double pitch = 1.0 + [=] {
+        if (pitchParameter < 0.0) {
+            return pitchParameter / 2.0;
+        } else {
+            return pitchParameter;
+        }
+    }();
 
     pState->m_pRubberBand->setPitchScale(pitch);
 
@@ -79,7 +100,8 @@ void PitchShiftEffect::processChannel(
             pInput,
             engineParameters.framesPerBuffer());
     pState->m_pRubberBand->process(
-            (const float* const*)pState->m_retrieveBuffer,
+            //static_cast<const float* const*>(pState->m_retrieveBuffer),
+            pState->m_retrieveBuffer,
             engineParameters.framesPerBuffer(),
             false);
 
@@ -88,7 +110,8 @@ void PitchShiftEffect::processChannel(
             framesAvailable,
             engineParameters.framesPerBuffer());
     SINT receivedFrames = pState->m_pRubberBand->retrieve(
-            (float* const*)pState->m_retrieveBuffer,
+            //static_cast<float* const*>(pState->m_retrieveBuffer),
+            pState->m_retrieveBuffer,
             framesToRead);
 
     SampleUtil::interleaveBuffer(pOutput,

--- a/src/effects/backends/builtin/pitchshifteffect.h
+++ b/src/effects/backends/builtin/pitchshifteffect.h
@@ -17,37 +17,17 @@ class RubberBandStretcher;
 
 class PitchShiftGroupState : public EffectState {
   public:
-    // This is the default increment from RubberBand 1.8.1.
-    static constexpr size_t kRubberBandBlockSize = 256;
+    PitchShiftGroupState(const mixxx::EngineParameters& engineParameters);
 
-    PitchShiftGroupState(const mixxx::EngineParameters& engineParameters)
-            : EffectState(engineParameters) {
-        initializeBuffer();
-        audioParametersChanged(engineParameters);
-    }
-
-    virtual ~PitchShiftGroupState();
-
-    void initializeBuffer() {
-        m_retrieveBuffer[0] = SampleUtil::alloc(MAX_BUFFER_LEN);
-        m_retrieveBuffer[1] = SampleUtil::alloc(MAX_BUFFER_LEN);
-    }
-
-    void audioParametersChanged(const mixxx::EngineParameters& engineParameters) {
-        m_pRubberBand = std::make_unique<RubberBand::RubberBandStretcher>(
-                engineParameters.sampleRate(),
-                engineParameters.channelCount(),
-                RubberBand::RubberBandStretcher::OptionProcessRealTime);
-
-        m_pRubberBand->setMaxProcessSize(kRubberBandBlockSize);
-        m_pRubberBand->setTimeRatio(1.0);
-    };
+    ~PitchShiftGroupState() override;
+    void initializeBuffer();
+    void audioParametersChanged(const mixxx::EngineParameters& engineParameters);
 
     std::unique_ptr<RubberBand::RubberBandStretcher> m_pRubberBand;
     CSAMPLE* m_retrieveBuffer[2];
 };
 
-class PitchShiftEffect : public EffectProcessorImpl<PitchShiftGroupState> {
+class PitchShiftEffect final : public EffectProcessorImpl<PitchShiftGroupState> {
   public:
     PitchShiftEffect() = default;
 

--- a/src/effects/backends/builtin/pitchshifteffect.h
+++ b/src/effects/backends/builtin/pitchshifteffect.h
@@ -20,7 +20,7 @@ class PitchShiftGroupState : public EffectState {
     PitchShiftGroupState(const mixxx::EngineParameters& engineParameters);
 
     ~PitchShiftGroupState() override;
-    void initializeBuffer();
+    void initializeBuffer(const mixxx::EngineParameters& engineParameters);
     void audioParametersChanged(const mixxx::EngineParameters& engineParameters);
 
     std::unique_ptr<RubberBand::RubberBandStretcher> m_pRubberBand;

